src/main/java/androidx/view/ViewGroup.kt

```diff
@@ -71,6 +71,16 @@ operator fun ViewGroup.iterator() = object : MutableIterator<View> {
     override fun remove() = removeViewAt(--index)
 }
 
+/** Returns a [Sequence] over the child views in this view group. */
+val ViewGroup.children: Sequence<View>
+    get() {
+        var index = -1
+        return generateSequence {
```

Comment:
This has quite a bit of overhead. Why don't we just return an `object : Sequence` whose function delegates to our existing `iterator()` method instead.

Reply:
would this be the same as something like:
``` kotlin
val ViewGroup.children: Sequence<View>
  get() = iterator().asSequence()
```

Reply:
I'd need to check what `asSequence` does. It probably prevents multiple iteration which my proposed solution doesn't.

Reply:
yeah it does, thats expected for a Sequence right?

Reply:
It's not required, no.